src/transport_ofi.c

```diff
@@ -603,24 +610,44 @@ static inline int exchange_and_av_insert(struct fabric_info *info)
 	return ret;
     }
 
+    full_ep_len = full_ep_len + epnamelen;
+    epnamebuff = malloc(full_ep_len);
+
     alladdrs = malloc(info->npes * epnamelen);
```

Comment:
yes, because the alladdrs array is just holding the first half of the epname value, which is the provider address used for avinsert. Essentially my method here is to piggy back on the av_insert PMI exchange so I've tagged on the hostname to the end of that array. I readjust the epname array based on whether or not you are using on_node_comms

I'm passing around epname which looks like this:

|___provider_address _from_getname__|____hostname____|
|--------epnamelen, updated by getname-|-HOSTNAME_MAX-|